crates/gradbench/src/main.rs

```diff
@@ -675,6 +701,18 @@ fn cli() -> Result<(), ExitCode> {
                 }
             }
         }
+        Commands::Log { command } => match command {
+            LogCommands::Trim { input, output } => {
+                let input_file = fs::File::open(input).map_err(|err| err_fail(anyhow!(err)))?;
+                let mut output_file: Box<dyn std::io::Write> = match output {
```

Comment:
Instead of using `dyn` here, could you just put separate calls to `log::trim` in the `Some` and `None` branches, like we do for `Commands::Run`? That way we avoid the unnecessary dynamic dispatch.

Reply:
For consistency with Unix conventions and ease of pipelining, I would also like to make the input option optional. Does this mean I should have four control paths? And if I understand the Rust compilation model, also four duplicated monomorphised functions?

Reply:
Correct.